### PR TITLE
Cmr 6901 cmr 6966 validate get data and file archive info formats in kms

### PR DESCRIPTION
**Pull request by sxu123**

### CODE REVIEW

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version/collection.clj [hunk 1]
```diff
@@ -2416,6 +2416,64 @@
         :MinimumValue 50,
         :MaximumValue 100}}]})
 
+(def sample-collection-1-15-5
```
Comment (mschmele) [inline umm-spec-lib/test/cmr/umm_spec/test/migration/version/collection.clj:2419]: Add empty case here as well

**Review by mschmele — COMMENTED** [review]
Just a few small tweaks - looks pretty good

File: ingest-app/src/cmr/ingest/validation/validation.clj [hunk 2]
```diff
@@ -101,7 +113,14 @@
      :ArchiveAndDistributionInformation
       {:FileDistributionInformation
        (match-kms-keywords-validation
-        kms-index :granule-data-format msg/data-format-not-matches-kms-keywords)}}))
+        kms-index :granule-data-format msg/data-format-not-matches-kms-keywords)
+       :FileArchiveInformation
+       (match-kms-keywords-validation
+        kms-index :granule-data-format msg/data-format-not-matches-kms-keywords)}
```
[the remark this answers is not in the document]
Comment (sxu123) [reply ingest-app/src/cmr/ingest/validation/validation.clj:119]: I just learned that you're not supposed to indent one space with a function call. :)

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj [hunk 3]
```diff
@@ -24,6 +25,23 @@
...
+  for RelatedUrls/GetData/Format in UMM-C 1.15.4 schema. For those values that don't exist in
+  the enum list, they will be mapped to \"Not provided\""
+  {"ascii" "ascii"
+   "binary" "binary"
+   "bufr" "BUFR"
+   "hdf4" "HDF4"
+   "hdf5" "HDF5"
+   "hdf-eos4" "HDF-EOS4"
+   "hdf-eos5" "HDF-EOS5"
+   "jpeg" "jped"
```
[the remark this answers is not in the document]
Comment (sxu123) [reply umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj:39]: Good catch! Thanks!

**Review by mschmele — APPROVED** [review]
Requested test cases shouldn't even be possible, so here's a green checkmark :)

File: umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj [hunk 4]
```diff
@@ -89,6 +107,15 @@
        (remove #(= "Military Grid Reference System" (:TilingIdentificationSystemName %)))
        seq))
 
+(defn- replace-invalid-format
+  "Replace all the GetData Format in RelatedUrls that doesn't exist in the 1.15.4 schema."
```
Comment (daniel-zamora) [inline umm-spec-lib/src/cmr/umm_spec/migration/version/collection.clj:111]: "Replace all the GetData Format in RelatedUrls that do not exist in the 1.15.4 schema."

File: ingest-app/src/cmr/ingest/validation/validation.clj (hunk 2 above)
[the remark this answers is not in the document]
Comment (daniel-zamora) [reply ingest-app/src/cmr/ingest/validation/validation.clj:119]: yea this formatting is busted.

[the remark this answers is not in the document]
Comment (daniel-zamora) [reply ingest-app/src/cmr/ingest/validation/validation.clj:119]: the diff view makes this look wrong, but I think its ok.

{code}
;; good
(filter
 even?
 (range 1 10))

(or
 ala
 bala
 portokala)

;; bad - two-space indent
(filter
  even?
  (range 1 10))

(or
  ala
  bala
  portokala)
{code}

**Review by daniel-zamora — APPROVED** [review]

**Pull request merged**